### DISCUSSION TITLE
file extensions are just noise

### DISCUSSION THREAD

**Discussion by nicholaswmin** [abbrcode/abbreviations-in-code#97] [Q&A]
Going down the path of including file extensions is going to create a substantial amount of noise for minimal value.

there is a minimal amount of grammatical space available and unnecessarily diluting it with (already well documented) file extensions doesn't make much sense.

I could be wrong. thoughts?

**Comment by T1xx1** [comment]
> I could be wrong. thoughts?

No, you could also be right.

JavaScript and TypeScript abbrs are inside the list as direct me effectively as a web dev and I thought they should be included (see past [#30](https://github.com/abbrcode/abbreviations-in-code/pull/30) and [#32](https://github.com/abbrcode/abbreviations-in-code/issues/32)) as they are not only file extensions (like .txt, .doc or .mp4) but programming file extensions (like .html, .css or .py)

Disclaimer:
> text • 🔴 txt

is included as a "normal" file extension but because is 🔴 Not recommended

My original thought was that maybe other kinds of programmers like C#, Swift, Rust or Go programmers (areas of programming not related to web dev) wouldn't know js or ts's file extensions (I don't have the minimum idea what Rust or Go's file extensions are they wouldn't stick in my brain even if I learned them simply because I don't care and I'm not using them).

I'm leaning towards your idea

---

> lovely project btw

Thank you. Just for clarity, I'm not the original author

**Comment by nicholaswmin** [comment]
> Thank you. Just for clarity, I'm not the original author

nice one, even more impressive.

Bit, I've upped the conciseness in my code style through the years and I've seen nothing but improvements. My brain just refuses to read verbose code anymore. 
Haskell's and the rest of the purists have their own abbrs that absolutely wouldn't work generically.

I'm a standards person and I can't be doing my own thing, nor expect anyone from my team to stick to my own abbreviations, so this project ticks boxes. Do you need any PR's for anything that's gonna keep the list nice and tidy?  Got a todo list I can maybe look at? Needs unit-tests in anything?

> but programming file extensions (like .html, .css or .py)

yes... I'm with you. Rust is `rs` and Go is `go`  but I get your point. Look, it's a non-problem right now so maybe leave as is and decide later. My position would be not to accept them readily anymore;


How do you use this? Manually look-up something on the site when you need it?

**Reply by T1xx1** [reply]
> > Thank you. Just for clarity, I'm not the original author
> 
> nice one, even more impressive.

Thanks again.

> Bit, I've upped the conciseness in my code style through the years and I've seen nothing but improvements. My brain just refuses to read verbose code anymore.

Same here.

> Haskell's and the rest of the purists have their own abbrs that absolutely wouldn't work generically.

This project is a general list of well-known abbrs and suggestions that could help people is called "Abbreviations in code" not "Abbreviations in Haskell code".

> Do you need any PR's for anything that's gonna keep the list nice and tidy?

I think we're in a good state.
Since I stumbled across this project, my [first pr](https://github.com/abbrcode/abbreviations-in-code/pull/13) aimed to increase the tidiness and readability of the mess this list was before introducing the dots of the degree of  recommendation and a general format we could follow (now this format is scripted and generated through code leaving no error for format issues)

> Got a todo list I can maybe look at?

Sure, all is public I'm not hiding anything https://github.com/orgs/abbrcode/projects

> > but programming file extensions (like .html, .css or .py)
> 
> yes... I'm with you. Rust is `rs` and Go is `go` but I get your point. Look, it's a non-problem right now so maybe leave as is and decide later. My position would be not to accept them readily anymore;

I'll think about this dilemma don't worry.
I think I'm the last maintainer here but I want leave this project I like it.

> How do you use this? Manually look-up something on the site when you need it?

Kinda

The original cause that brought me to this project was a VSCode extension that I built that would help me manage abbrs in my codebase. Highlighting recommended words to abbreviate and highlight abbrs which were unclear or could have duplicate meanings.

I kinda achieved my goal https://github.com/T1xx1/AbbrExt but I still decided to build the site so we could have a search function.
I think is nice to have a mini ecosystem working: it's your choice If you want to read the list manually, copy it somewhere you can access easily (I'm using this method with my Obsidian vault), search with the site or use the extension in your editor.

Sorry if I'm unclear English is not my first language feel free to ask for clarifications

**Reply by nicholaswmin** [reply]
Nope everything is crystal clear. Thank you very much for taking the time to reply. I'll drop by again soon

**Comment by T1xx1** [comment]
https://github.com/abbrcode/abbreviations-in-code/commit/f0f3a4407477bb48092990f57791bb8d73eec3a4
